src/App.tsx

```diff
@@ -1,24 +1,26 @@
+import {
+	BrowserRouter as Router,
+	Routes,
+	Route,
+	Navigate,
+} from 'react-router-dom';
 import { Header } from './components/Header/Header';
-import { SearchBar } from './components/Courses/components/SearchBar/SearchBar';
+import { Login } from './components/Login/Login';
+import { Registration } from './components/Registration/Registration';
 import { Courses } from './components/Courses/Courses';
-import styles from './App.module.css';
-import { Button } from './common/Button/Button';
+import { CourseInfo } from './components/CourseInfo/CourseInfo';
 import { CreateCourse } from './components/CreateCourse/CreateCourse';
+import { AuthComponent } from './components/AuthComponent/AuthComponent';
 import { useMemo, useState } from 'react';
-import { mockedAuthorsList, mockedCoursesList, ICourse } from './mockData';
+import { mockedAuthorsList, mockedCoursesList } from './mockData';
 import { v4 as uuidv4 } from 'uuid';
 
 function App() {
 	const [courses, setCourses] = useState(mockedCoursesList);
 	const [authors, setAuthors] = useState(mockedAuthorsList);
 	const [search, setSearch] = useState('');
-	const [isCourses, setIsCourses] = useState(true);
 	const [showSearchCourses, setShowSearchCourses] = useState(false);
 
-	const onClickAddNewCourse = () => {
-		setIsCourses(false);
-	};
-
 	const handleSearchInput = (value: string) => {
```

Comment:
if a component handles routing - it shouldn't do anything else, no passing props, no state hooks, etc, reason: separation of concerns, which would be very visible when unit testing this component - mocking router when testing a search input would feel very uneasy
`<Route path='registration' element={<Registration />} />` is what every route should look like